File: depyler_tests/test_013_enter_returns_self.py
class Connection:
    def __init__(self, host: str):
        self.host = host
        self.connected = False

    def __enter__(self):
        self.connected = True
        return self

    def __exit__(self, exc_type, exc_val, exc_tb):
        self.connected = False
        return False

    def query(self, sql: str) -> str:
        return f"Result from {self.host}"

def main():
    with Connection("localhost") as conn:
        result = conn.query("SELECT 1")
        print(result)

File: depyler_tests/test_013_enter_returns_self.rs
use serde_json;
#[derive(Debug, Clone)]
pub struct Connection {
    pub host: String,
    pub connected: bool,
}
impl Connection {
    pub fn new(host: String) -> Self {
        Self {
            host,
            connected: false,
        }
    }
    pub fn __enter__(&mut self) -> &Self {
        self.connected = true;
        return self;
    }
    pub fn __exit__(
        &mut self,
        exc_type: serde_json::Value,
        exc_val: serde_json::Value,
        exc_tb: serde_json::Value,
    ) -> bool {
        self.connected = false;
        return false;
    }
    pub fn query(&self, sql: String) -> String {
        return format!("Result from {}", self.host);
    }
}
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn main() {
    let _context = Connection::new("localhost".to_string().to_string());
    let conn = _context.__enter__();
    let result = conn.query("SELECT 1".to_string());
    println!("{}", result);
}
